### PR TITLE
Fix prometheus role nfs

### PR DESCRIPTION
**Pull request by zgalor**
Allow for external or internal nfs.
Use facts as used for logging and metrics.
Also updated alertmanager image default tag to a stable version

### CODE REVIEW

**Comment by zgalor** [comment]
@ewolinetz I am still testing this, but I'd like to get your review, so I can fix issues before dev-cut.
Take your time. I'll be on holiday till EOW.

**Comment by ewolinetz** [comment]
So far, so good. I think you're missing changes in `filter_plugins/oo_filters.py` too. They're more apparent in https://github.com/openshift/openshift-ansible/pull/5176

File: filter_plugins/oo_filters.py [hunk 1]
```diff
@@ -985,6 +997,18 @@ def oo_persistent_volume_claims(hostvars, persistent_volume_claims=None):
         persistent_volume_claim = oo_component_pv_claims(hostvars, 'metrics')
         if persistent_volume_claim is not None:
             persistent_volume_claims.append(persistent_volume_claim)
+    if 'prometheus' in hostvars['openshift']:
```
Comment (smarterclayton) [inline filter_plugins/oo_filters.py]: @sdodson this sort of duplication in core modules is exactly what my underlying concern is with the coupling between our different components.  Having to do this is a code smell and is coupling completely unrelated to playbooks.

File: roles/openshift_storage_nfs/tasks/main.yml [hunk 2]
```diff
@@ -35,6 +35,9 @@
     - "{{ openshift.logging }}"
     - "{{ openshift.loggingops }}"
     - "{{ openshift.hosted.etcd }}"
+    - "{{ openshift.prometheus }}"
```
Comment (smarterclayton) [inline roles/openshift_storage_nfs/tasks/main.yml]: The fact that this role knows about all possible things that could ever use NFS is a serious architectural coupling problem.

(zohar not picking on you, picking on where we need to change course in ansible)

**Comment by ewolinetz** [comment]
@zgalor looks like all the tests passed. Do you want to remove the WIP in the title and squash? Then we can give it a lgtm

**Comment by zgalor** [comment]
@ewolinetz tempting... But I want to run some more tests and also update the examples, and then I'll remove the WIP

**Comment by sdodson** [comment]
@zgalor still more testing you'd like to do before we accept this?

**Comment by zgalor** [comment]
@sdodson this is not ready yet. I need to add documentation, and some more testing.
I'm on PTO this week so it won't advance. I'll remove the wip next week and tag you once done

**Comment by sdodson** [comment]
Ok thanks.

**Comment by zgalor** [comment]
@sdodson @ewolinetz now I finally tested everything, squashed and removed the WIP.
Please re-review and if all tests pass you can merge.
Not much changed since your last review. Just rebased, added examples and updated the image tag for alertmanager.

File: inventory/byo/hosts.example [hunk 3]
```diff
@@ -615,6 +615,71 @@ openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true',
...
+# Prometheus storage config
+# Option A - NFS Host Group
+# An NFS volume will be created with path "nfs_directory/volume_name"
+# on the host within the [nfs] host group.  For example, the volume
+# path using these options would be "/exports/prometheus"
+#openshift_prometheus_storage_kind=nfs
+#openshift_prometheus_storage_access_modes=['ReadWriteOnce']
+#openshift_prometheus_storage_nfs_directory=/exports
+#openshift_prometheus_storage_nfs_options='*(rw,root_squash)'
+#openshift_prometheus_storage_volume_name=prometheus
```
Comment (tbielawa) [inline inventory/byo/hosts.example]: README requires variable documentation updates to describe all the new config parameters 

If some of these are not intended to be overridden by the users (e.g., *constants*) then they should be moved into the `vars/` directory and may not require additional documentation.

**Comment by sdodson** [comment]
We should port this over to using the openshift_nfs role in the future but accept it for now.

**Comment by tbielawa** [comment]
@sdodson and I just spoke, I'm going to lgtm this again so long as we can get the documentation addressed asap in a follow-up PR.

File: inventory/byo/hosts.example (hunk 3 above)
Comment (zgalor) [reply inventory/byo/hosts.example]: @tbielawa I started changing the readme but then I realized these variables are not in this role scope, and shouldn't be added in the role README.
If any, this README should update:
https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_storage_nfs
but, it currently doesn't have all the logging, metrics, etc vars in it.

Was that the one you meant?

**Pull request merged**